stolostron/backlog#18755 upd hypershift (#3613)
open-cluster-management/rhacm-docs: clusters/hosted_control_plane_configure.adoc

> [#hosted-control-plane-configure]

Reviewer: ```suggestion
[#hosted-control-planes-configure]
```

= Configuring hosted control planes

Hosted control planes is a Technology Preview feature, so the related components are disabled by default. Enable the feature by editing the `multiclusterengine` custom resource to set the `spec.overrides.components[?(@.name=='hypershift-preview')].enabled` to `true`. 

> Enter the following command to verify that the hosted control planes feature is enabled:

Reviewer: ```suggestion
Enter the following command to ensure that the hosted control planes feature is enabled:
```

Reviewer: The is because `patch` mutates already.

----
oc patch mce multiclusterengine-sample--type=merge -p '{"spec":{"overrides":{"components":[{"name":"hypershift-preview","enabled": true}]}}}'
----

[#hosting-service-cluster-configure]

> == Using an existing cluster as a hosting service cluster

Reviewer: Coming across this before being introduced, I'd ask what is a hosting service cluster?

Reviewer: We can define at first occurence like here or link to somewhere

Author: Added some description.

> You can deploy hosted control planes by configuring an existing cluster to function as a hosting service cluster, and then deploying a cluster that is managed by the HyperShift hosting service cluster that you enabled as the hosting service cluster. The hosting service cluster can be the {product-title-short} hub cluster or one of the {ocp-short} managed clusters.

Reviewer: > and then deploying a cluster that is managed by the HyperShift hosting service cluster that you enabled as the hosting service cluster
What are we trying to say here?

[#hosting-service-cluster-configure-prereq]
=== Prerequisites

> You must have the multicluster engine operator installed on at least one cluster that is managed by {ocp}. The multicluster engine operator is automatically installed when you install {product-title-short} version 2.5, and later.

Reviewer: ```suggestion
You must have the multicluster engine operator installed on at least one cluster that is managed by {ocp}. The multicluster engine operator is automatically installed when you install {product-title-short} version 2.5, and later but can be installed independently from {product-title-short} as an OpenShift operator from the operator hub.
```

Author: Updated to clarify that it can be installed without ACM.

> *Note:* If you want your {product-title-short} hub cluster to be your hosting service cluster, you can configure `local-cluster` as your hosting service cluster. A hub cluster that is also managed by itself is designated as `local-cluster` in the list of clusters. 

Reviewer: How we do so is missing.

```suggestion
*Note:* If you want your {product-title-short} hub cluster to be your hosting service cluster, you can configure `local-cluster` as your hosting service cluster. A hub cluster that is also managed by itself is designated as `local-cluster` in the list of clusters. 

To use the hub cluster (`local-cluster`) as the hosting service cluster, you can import it as follows:

$ oc apply -f - <<EOF
apiVersion: cluster.open-cluster-management.io/v1
kind: ManagedCluster
metadata:
  labels:
    local-cluster: "true"
  name: local-cluster
spec:
  hubAcceptsClient: true
  leaseDurationSeconds: 60
EOF
```

[#hosted-control-plane-configure]
= Configuring hosted control planes

Hosted control planes is a Technology Preview feature, so the related components are disabled by default. Enable the feature by editing the `multiclusterengine` custom resource to set the `spec.overrides.components[?(@.name=='hypershift-preview')].enabled` to `true`. 

Enter the following command to ensure that the hosted control planes feature is enabled:

----
oc patch mce multiclusterengine-sample--type=merge -p '{"spec":{"overrides":{"components":[{"name":"hypershift-preview","enabled": true}]}}}'
----

[#hosting-service-cluster-configure]
== Using an existing cluster as a hosting service cluster

> You can deploy hosted control planes by configuring an existing cluster to function as a hosting service cluster, and then deploying a cluster that is managed by the HyperShift hosting service cluster that you enabled as the hosting service cluster. The hosting service cluster is the {ocp-short} cluster where the control planes are hosted, and can be the {product-title-short} hub cluster or one of the {ocp-short} managed clusters.

Reviewer: {product-title-short} What will this translate to?

Author: That will resolve to Red Hat Advanced Cluster Management.

Reviewer: What if we are using MCE, is it called the MCE hub cluster?

Author: Good question. I removed {product-title-short}, as both MCE and ACM have a hub cluster. This way, it can be either one. Thanks!

kind: ManagedClusterAddOn
metadata:
  name: hypershift-addon

>   namespace: <managed-cluster-name> 

Reviewer: `<managed-cluster-name>  # e.g. local-cluster if hub was used as the hosting service cluster`

Author: This was fixed later.

spec:
  installNamespace: open-cluster-management-agent-addon
----
+

> Replace `managed-cluster-name` with the name of the managed cluster on which you want to install the HyperShift operator.

Reviewer: ```suggestion
Replace `managed-cluster-name` with the name of the managed cluster on which you want to install the HyperShift operator. If you used `local-cluster`, then the `managed-cluster-name` should be `local-cluster`
```

+
Replace `filename` with the name of the file that you created. 

> . Create an OIDC S3 credentials secret named `hypershift-operator-oidc-provider-s3-credentials` for the HyperShift operator. Create the secret in the namespace of the cluster where the hosting service cluster is hosted. 

Reviewer: ```suggestion
Since hosted control planes is available as a technology preview only on AWS,  you can start by creating an OIDC S3 credentials secret for the HyperShift operator, and name it `hypershift-operator-oidc-provider-s3-credentials`. It should reside in managed cluster namespace (i.e., the namespace of the managed cluster that will be used as the hosting service cluster). If you used `local-cluster`, then create the secret in the `local-cluster` namespace
```

Author: Updated this content to include suggestions.
